torch/_dynamo/eval_frame.py

```diff
@@ -123,7 +123,7 @@ def enable_dynamic(enable: bool = True):
         return
     with patch("torch._dynamo.config.dynamic_shapes", True), patch(
         "torch._functorch.config.use_dynamic_shapes", True
-    ):
+    ), patch("torch._dynamo.config.specialize_int_float", False):
```

Comment:
It predates dynamic shapes, I think? I think there is a bigger refactor that probably can be done here.

Reply:
Hm, is it safe? We do want to specialize on floats sometimes (e.g. dropout probability, or some control flow due to int/float arg)

Reply:
If we control flow on int/float, this will cause a specialization anyway. I think this rolls into the more general problem which is what should the dynamic shapes UI be; I think dynamic=True is probably not the right terminal state.

Reply:
How does this work if this int is passed into the shape to construct a tensor ? dont we need to specialize ?